torch_geometric/transforms/pad.py

```diff
@@ -0,0 +1,548 @@
+                    assert (src_node in self.max_num_nodes.value), \
+                        f'Padding for edge {type} was specified, while the '\
+                        f'src node of type {src_node} features were not '\
+                        'padded. This means the new padded edges cannot be '\
+                        'added as they would point from non-existent padded '\
+                        'src node.'
+
+                    assert (dst_node in self.max_num_nodes.value),\
+                        f'Padding for edge {type} was specified, while the '\
+                        f'dst node of type {dst_node} features were not '\
+                        'padded. This means the new padded edges cannot be '\
+                        'added as they would point to non-existent padded '\
+                        'dst node.'
+
+                # The edges are stored in a form of tensor of shape
+                # [2, 'num of edges'], so we pick dim = 1.
+                num_edges = value.shape[1]
+
+                if self.max_num_edges.is_number:
+                    max_num_edges = self.max_num_edges.value
+                else:
+                    type_key = (type[0], type[2])
+                    if type_key in self.max_num_edges.value:
+                        if self.max_num_edges.value[type_key].is_number:
+                            max_num_edges = self.max_num_edges.value[type_key]
+                        else:
+                            if edge_type in self.max_num_edges.value[
+                                    type_key].value:
+                                max_num_edges = self.max_num_edges.value[
+                                    type_key].value[edge_type]
+                            else:
+                                # The type (src_node, edge_type, dst_node)
+                                # is NOT included in the user's input, which
+                                # means the user decided NOT to pad the
+                                # edge type.
+                                continue
+                    else:
+                        # The type (src_node, edge_type, dst_node) is NOT
+                        # included in the user's input, which means the user
+                        # decided NOT to pad the edge type.
+                        continue
+
+                assert num_edges <= max_num_edges, \
+                    f'Too many edges. Graph has {num_edges} edges of type '\
+                    f'{type} defined and max_num_edges is '\
+                    f'{self.max_num_edges.value}. ' \
+                    'The max_num_edges option must be at least as large '\
+                    'as\n\n' \
+                    '    max(data.num_edges)\n\n' \
+                    'for all examples in the dataset.'
+
+    @singledispatchmethod
+    def __call__(self, data):
+        raise ValueError(f'Unsupported data type: {type(data)}')
+
+    @__call__.register
```

Comment:
I personally think the implementation gets cleaner if you merge both `Data` and `HeteroData` handling into a single function and just iterate over its stores. There is a lot of functionality to be shared here.

Reply:
Will try to unify that tomorrow